pkgs/development/tools/rust/itm/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "itm";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "rust-embedded";
+    repo = "itm";
+    rev = "96d7710f2ba8f915668e95d67e0def961567c195";
+    sha256 = "15pa0ydm19vz8p3wairpx3vqzc55rp4lgki143ybgw44sgf8hraj";
+  };
+
+  cargoSha256 = "1d748vh1ig8r0cd70lsnw9fdmanic4jxxibbbagd5mrh8462v64v";
+  cargoPatches = [ ./cargo_lock.patch ];
```

Comment:
What I was trying to say, is that it seems that upstream is out-of-date. The proper fix would be to upstream the lock update, then use `fetchpatch` to retrieve it from the PR. Instead of vendoring the patch into the nix expression.

Reply:
So, after some digging I've learned that this tool only exists up to the 0.3.1 release of this tool. I want to package up this tool to make it easier for others following along with the Rust-Embedded Discovery using NixOS, but it looks like the CLI tool I'm trying to package has been extracted into another non-compatible tool [here](https://github.com/japaric/itm-tools).

The book still calls for this tool to be used (see https://docs.rust-embedded.org/discovery/03-setup/index.html#itmdump) so I think it's still appropriate to package this up at least until the Discovery book is reworked to reference this other tool.

Reply:
Uh, so, as for fixing this upstream, this tool isn't actually around anymore so there's nothing to fix upstream 😓

Reply:
hmm, if it's no longer maintained, then I'm not really comfortable adding it to nixpkgs. because all ownership of maintenance will be put on us.

Reply:
Very fair. I will see what I can do :)